libraries/AP_Generator/AP_Generator_RichenPower.cpp

```diff
@@ -194,6 +194,23 @@ constexpr float AP_Generator_RichenPower::heat_required_for_run()
 }
 
 
+void AP_Generator_RichenPower::check_maintenance_required()
+{
+    // don't bother the user while flying:
+    if (hal.util->get_soft_armed()) {
+        return;
+    }
+
+    const uint32_t now = AP_HAL::millis();
+
+    if (last_reading.errors & (1U<<uint16_t(Errors::MaintenanceRequired))) {
+        if (now - last_maintenance_warning_ms > 5000) {
```

Comment:
change to once a minute

Reply:
add GEN_OPTIONS bit to suppress the warning

Reply:
Both of these have been done and tested in SITL.